code/web/src/modules/survey/Survey.js

```diff
@@ -0,0 +1,120 @@
+// Imports
+import React, { useEffect, useState } from 'react';
+import { APP_URL } from '../../setup/config/env';
+import { connect } from 'react-redux';
+import { withRouter } from 'react-router-dom'
+
+// App Imports
+import { surveyQuestions } from './surveyQuestions';
+import userRoutes from '../../setup/routes/user';
+import { postUserStyle } from './api/actions';
+import { login } from '../user/api/actions';
+
+const Survey = (props) => {
+  const [question, setQuestion] = useState(surveyQuestions[0]);
+  const [userStyle, setUserStyle] = useState(null);
+  const [styleAnswers, setStyleAnswers] = useState({
+    Grunge: [],
+    Sporty: [],
+    Classic: [],
+    Casual: [],
+    Bohemian: [],
+  });
+
+  const recordAnswer = (cat) => {
+    setStyleAnswers(styleAnswers, styleAnswers[cat].push(question.id));
+    let i = surveyQuestions.indexOf(question);
+
+    if (i === surveyQuestions.length - 1) {
+      calculateStyle();
+      return;
+    }
+
+    setQuestion(surveyQuestions[i + 1]);
+  };
+
+  const calculateStyle = () => {
+    let tally = Object.entries(styleAnswers);
+    let sortedStyles = tally.sort((a, b) => {
+      return b[1].length > a[1].length ? 1 : -1;
+    });
+
+    let style1 = sortedStyles[0][1].length;
+    let style2 = sortedStyles[1][1].length;
+
+    if (style1 === 1) {
+      setUserStyle('Ecletic');
+    } else if (style1 === style2) {
+      setUserStyle(`${sortedStyles[0][0]} but ${sortedStyles[1][0]}`);
+    } else if (style1 > style2) {
+      setUserStyle(`${sortedStyles[0][0]}`);
+    }
+  };
```

Comment:
Still wondering if there is a cleaner way to do the style calculation, lets revisit in the future